### DISCUSSION TITLE
prevent person alerts from clouds moving across car windshield

### DISCUSSION THREAD

**Discussion by stoneobscurity** [blakeblackshear/frigate#8706] [General]
need some help:
i keep getting false alerts on clouds on car windshield.

![snapshot_360](https://github.com/blakeblackshear/frigate/assets/5900573/567a2cd5-8291-4bbc-801a-55f8a702b8d2)

my person threshold is set to 73, and these event are showing 75-85.
if i mask/filter it, then when the car is not there, it would not alert on a person in my driveway.

![Screenshot 2023-11-18 at 1 35 13 PM](https://github.com/blakeblackshear/frigate/assets/5900573/5d841ff2-011d-4ac4-a82b-00f9ce101922)

(frig needs a 'if car present, active filter, otherwise, disable filter')

**Comment by NickM-27** [comment]
I would suggest a max_ratio filter might work well here

**Reply by stoneobscurity** [reply]
i don't see how. the clouds are not static, they are in constant motion. any filter i do here would get in the way when the car is not there.

![clouds](https://github.com/blakeblackshear/frigate/assets/5900573/0caccf73-8e2f-443e-8394-b57ee30a3137)

**Reply by NickM-27** [reply]
A person shouldn't have a ratio that wide regardless of if a car is there or not.

**Reply by stoneobscurity** [reply]
but they could be that height. it'll filter on both.

![snapshot](https://github.com/blakeblackshear/frigate/assets/5900573/e0049e4b-0143-4660-bf81-913765c14e48)

**Reply by NickM-27** [reply]
Ratio filter has nothing to do with object height directly. It's about the ratio of width to height.

**Reply by stoneobscurity** [reply]
that is not what the docs say.

# Optional: maximum width/height of the bounding box for the detected object (default: 24000000)

**Reply by NickM-27** [reply]
It means maximum width divided by height, which is the ratio. This is explained further in the beta docs https://deploy-preview-6262--frigate-docs.netlify.app/guides/false_positives#object-proportions

**Reply by stoneobscurity** [reply]
i found a better explanation on reddit those docs.

"the ratio is width / height so 1 is a square, < 1 is a box that is taller than wide and > 1 is a box that is wider than it is tall."

so what min/max would you recommend?

**Reply by NickM-27** [reply]
Maybe try 1.0 and see how that works.

**Reply by MrSiO** [reply]
Any updates on that issue?
If you could share the final ratio config.

I have the same issue and I applied a min_ratio to "person" and "dog".

I've set 0.5 to person under detect Objects/Filters/Person/min_ratio (so person should have a hight at least twice of its width) but it's not working at the moment. I'm on beta 13.0-7. As if ratio was not taken into consideration.

Thx for any updates.

**Reply by NickM-27** [reply]
I responded to your issue, you're most likely using the wrong ratio filter.

**Reply by MrSiO** [reply]
> I responded to your issue, you're most likely using the wrong ratio filter. 

And you are absolutely right.

So with a max_ratio of 0.5 for Person, users should get rid of most of those false positive from cloud reflections in windows/windshields.

Nice feature 👍
